Index: src/com/google/gwt/gen2/table/override/client/HTMLTable.java
===================================================================
--- src/com/google/gwt/gen2/table/override/client/HTMLTable.java	(revision 1673)
+++ src/com/google/gwt/gen2/table/override/client/HTMLTable.java	(working copy)
@@ -15,6 +15,8 @@
  */
 package com.google.gwt.gen2.table.override.client;
 
+import com.google.gwt.dom.client.Node;
+import com.google.gwt.dom.client.NodeList;
 import com.google.gwt.user.client.DOM;
 import com.google.gwt.user.client.Element;
 import com.google.gwt.user.client.Event;
@@ -437,6 +439,16 @@
      */
     private void prepareColumnGroup() {
       if (columnGroup == null) {
+        NodeList<Node> childs = tableElem.getChildNodes();
+        if (childs != null) {
+          for (int i = 0, n = childs.getLength(); i < n; i++) {
+            Element node = childs.getItem(i).cast();
+            if (node.getTagName().equalsIgnoreCase("colgroup")) {
+              columnGroup = node;
+              return;
+            }
+          }
+        }
         columnGroup = DOM.createElement("colgroup");
         DOM.insertChild(tableElem, columnGroup, 0);
         DOM.appendChild(columnGroup, DOM.createElement("col"));
